### ISSUE TITLE
Tree view BG color makes it unreadable

### ISSUE THREAD

**Issue by vrelk** [BlueMystical/Dark-Mode-Forms#58]
It seems that if you modify a TreeView while the program is running (ex. progressively loading a remote file tree), as soon as you deselect one of the newly updated nodes, the coloring changes to white, making it unreadable until you select it again.

![image](https://github.com/user-attachments/assets/a91ad0e3-7643-408b-b0fd-3c5f96e09a4a)
In this case, I currently have "firmware" selected, and the white blocks are ones that were previously selected. I tried manually setting the background color to something else (ex. Blue). When it loads the first time, it's blue as expected, but as soon as you deselect it, it turns white.

Reapplying the theme, or using Refresh() and Update() functions on the TreeView doesn't appear to help.

**Comment by kachnitata** [comment]
I use a couple of `Treeview`s that change dynamically but i have never had this issue. Are you sure you don't interfere with the style in your code? Would you mind sending your `Treeview`'s initialization code from there .Designer file, so that i can compare it with mine?

**Comment by kachnitata** [comment]
Actually... I realized I always deselect the current node before changing the tree and then select it again. That's probably why i never run into this issue.

**Comment by kachnitata** [comment]
Another difference night be that my changes to the tree are always synchronously triggered from the UI, never by any other thread. Are you using `Invoke` when updating the tree?

**Comment by kachnitata** [comment]
Plus I use `BeginUpdate`/`EndUpdate` , i don't know if it may play any role...

**Comment by BlueMystical** [comment]
Unfortunately i have not been able to reproduce this behavior:
i am dynamically loading the structure of the 'My documents' folder in the TreeView using this code:
```csharp
		//Agregar estructura de archivos y directorios recursivamente:
		private void ListDirectory(TreeView treeView, string path, string filter)//<- filter = "*.exe;*.jpg"
		{
			treeView.Nodes.Clear();
			DirectoryInfo rootDirectoryInfo = new DirectoryInfo(path);
			treeView.Nodes.Add(CreateDirectoryNode(rootDirectoryInfo, filter));
		}
		private static TreeNode CreateDirectoryNode(DirectoryInfo directoryInfo, string filter)
		{
			TreeNode directoryNode = new TreeNode(directoryInfo.Name);
			foreach (DirectoryInfo directory in directoryInfo.GetDirectories())
			{
				try
				{
					directoryNode.Nodes.Add(CreateDirectoryNode(directory, filter));
				}
				catch { }
			}

			foreach (FileInfo file in directoryInfo.GetFiles(filter))
			{ 
				try
				{
					directoryNode.Nodes.Add(new TreeNode(file.Name));
				}
				catch { }
			}				
			return directoryNode;
		}
```
```csharp
string myDocumentsPath = Environment.GetFolderPath(Environment.SpecialFolder.MyDocuments);
ListDirectory(this.treeView1, myDocumentsPath, "*.*");
```
But the control behaves as intended:
![image](https://github.com/user-attachments/assets/78a2825e-bd25-450c-a5c6-c315a5fa66a6)

Would you mind to share your code? or a way to replicate this?
Thanks.

**Comment by vrelk** [comment]
I was working on an example program for you, but found that I was unable to duplicate the issue. While copying chunks of code over, I found a stray `node.BackColor = Color.White;` that was leftover from when I was doing things with the default theme. I apologize for wasting your time trying to duplicate the issue, but it would appear that the issue was me. (I spent way too much time trying to track this down the other day, but never noticed that line of code.)
